source/common/network/address_impl.cc

```diff
@@ -247,6 +257,13 @@ std::string Ipv6Instance::Ipv6Helper::makeFriendlyAddress() const {
   char str[INET6_ADDRSTRLEN];
   const char* ptr = inet_ntop(AF_INET6, &address_.sin6_addr, str, INET6_ADDRSTRLEN);
   ASSERT(str == ptr);
+  if (address_.sin6_scope_id != 0) {
+    char if_name[IF_NAMESIZE]{'\0'};
+    if (if_indextoname(address_.sin6_scope_id, if_name) != nullptr) {
```

Comment:
This is will create a performance bottleneck if config has a lot of link local IPv6 addresses. We had to revert a PR that was resolving interface name for logging some time ago, because `if_indextoname` can take time to complete. This is not a problem right now, since such configs do not exist.

While I think it is nice to have the interface name instead of index in the friendly name, I do not think it is required. I think index would do just fine. My suggestion is to just use index for now and avoid any potential timing issues with calling `if_indextoname`.

Reply:
done and added a comment to capture the above.